library/std/src/ffi/c_str.rs

```diff
@@ -1299,7 +1319,13 @@ impl CStr {
     #[stable(feature = "cstr_from_bytes", since = "1.10.0")]
     #[rustc_const_unstable(feature = "const_cstr_unchecked", issue = "none")]
     pub const unsafe fn from_bytes_with_nul_unchecked(bytes: &[u8]) -> &CStr {
-        &*(bytes as *const [u8] as *const CStr)
+        // SAFETY: `bytes` is a reference and so safe to cast to a pointer.
```

Comment:
```suggestion
        // SAFETY:
```

Creating a raw pointer is always safe.